### PR TITLE
Document HTTP Status Codes

### PR DESCRIPTION
**Pull request by lidel**
This PR:

- Adds section on HTTP Status Codes
  as suggested in https://github.com/ipfs/go-ipfs/issues/6746#issuecomment-549759085 by @Stebalien 
- Adds "HTTP Commands" header, which was missing
- Switches to go-ipfs v0.4.22
- Fixes build under go v1.13.4 (not sure if it can be done better, but #worksForMe :upside_down_face:)

(i will PR to docs.ipfs.io and  link to a preview shortly)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
-	github.com/ipfs/go-ipfs v0.4.21
+	github.com/ipfs/go-ipfs v0.4.22
 	github.com/ipfs/go-ipfs-cmds v0.0.8
 	github.com/libp2p/go-libp2p-peer v0.1.1
 	github.com/libp2p/go-libp2p-peerstore v0.0.6
 	github.com/multiformats/go-multiaddr v0.0.4
+	golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898 // indirect
 )
+
+replace github.com/go-critic/go-critic v0.0.0-20181204210945-ee9bf5809ead => github.com/go-critic/go-critic v0.3.5-0.20190210220443-ee9bf5809ead
```
Comment (hsanjuan) [inline go.mod]: This looks pretty horrible

**Comment by hsanjuan** [comment]
Works for me without hacking the imports

**Comment by hsanjuan** [comment]
Nevermind, it seems I'm running a go 1.13 version that still does not fuck up everything.

**Comment by jessicaschilling** [comment]
@lidel @hsanjuan What was your workaround for the CircleCI build test failing? I'd like to merge https://github.com/ipfs/docs/pull/368 ...

File: markdown.go [hunk 2]
```diff
@@ -104,7 +108,21 @@ flag is the %s query parameter below:
...
+### HTTP Status Codes
+
+Status codes used at the RPC layer are simple:
+
+* `+"`500`"+` - RPC endpoint returned an error
+* `+"`404`"+` - RPC endpoint doesn't exist
+* `+"`400`"+` - Malformed RPC, argument type error, etc
+* `+"`403`"+` - RPC call forbidden
+
+In other words, `+"`500`"+` means that the function _does_ exist, it just failed internally for some reason. To know that reason, you have to look at the "application layer" error (commands lib error).
```
Comment (Stebalien) [inline markdown.go]: note: functions can _also_ fail with a 200 status code by later returning a trailing error. :sob:.

**Pull request merged**